### PR TITLE
Fix fields handling to properly batch HTTP request payloads

### PR DESCRIPTION
**Pull request by rvmiller89**
Noticed by Paul, after adding the additional `fields` to `_sumo_metadata`, we were not properly batching the HTTP request payloads in order to set `X-Sumo-Fields`.

This PR adds handling for `fields` from `_sumo_metadata` and adds test cases for:

1. Batching without header/field overrides
2. Batching with header overrides
3. Batching with field overrides

Some of this may conflict with https://github.com/SumoLogic/fluentd-output-sumologic/pull/53 but we can reconcile the differences.

### CODE REVIEW

File: lib/fluent/plugin/out_sumologic.rb [hunk 1]
```diff
@@ -195,36 +195,20 @@ def sumo_key(sumo_metadata, chunk)
...
-    fields = sumo_metadata['fields'] || ""
-    Hash[
-        fields.split(',').map do |pair|
-          k, v = pair.split('=', 2)
-          [k, v]
-        end
-    ]
-  end
-
-  def dump_collected_fields(log_fields)
```
Comment (rvmiller89) [inline lib/fluent/plugin/out_sumologic.rb]: this function will likely come back after we do https://github.com/SumoLogic/fluentd-output-sumologic/pull/53

File: lib/fluent/plugin/out_sumologic.rb [hunk 2]
```diff
@@ -195,36 +195,20 @@ def sumo_key(sumo_metadata, chunk)
     source_host = sumo_metadata['host'] || @source_host
     source_host = extract_placeholders(source_host, chunk) unless source_host.nil?
 
-    "#{source_name}:#{source_category}:#{source_host}"
+    fields = sumo_metadata['fields'] || ""
+    fields = extract_placeholders(fields, chunk) unless fields.nil?
+
+    "#{source_name}:#{source_category}:#{source_host}:#{fields}"
```
Comment (rvmiller89) [inline lib/fluent/plugin/out_sumologic.rb]: not addressed by this PR, but something we should fix: having a colon (`:`) in any metadata field or value would currently break this

Comment (samjsong) [reply lib/fluent/plugin/out_sumologic.rb]: good point - I believe you can use ruby hashes as keys? maybe that's something we can look into though it does make it pretty complex

**Comment by rvmiller89** [comment]
Yeah these are all great points, I am thinking start with this for now, and revisit our HTTP batching to account for these edge cases.

**Pull request merged**